crypto/eddsa.go

```diff
@@ -0,0 +1,78 @@
+package crypto
+
+import (
+	"bytes"
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+)
+
+// Ed25519GenerateKey returns a random PEM encoded Ed25519 Private Key.
+func Ed25519GenerateKey() ([]byte, error) {
+	_, secret, err := ed25519.GenerateKey(rand.Reader)
+	if err != nil {
+		return nil, fmt.Errorf("Ed25519GenerateKey: failed to generate EDDSA private key: %w", err)
+	}
+	return pemEncodeEdPrivateKey(secret)
+}
+
+// Ed25519GenerateKeyFromSeed returns a PEM encoded Ed25519 Private Key from
+// `seed`. Panics if len(seed) is not SeedSize.
+func Ed25519GenerateKeyFromSeed(seed []byte) ([]byte, error) {
+	return pemEncodeEdPrivateKey(ed25519.NewKeyFromSeed(seed)) // Panics
+}
+
+// Ed25519DerivePublicKey returns an ed25519 Public Key from given PEM encoded
+// `privatekey`.
+func Ed25519DerivePublicKey(privatekey []byte) ([]byte, error) {
+	secret, err := ed25519DecodeFromPEM(privatekey)
+	if err != nil {
+		return nil, fmt.Errorf("EDDSADerivePublicKey: could not decode private key")
```

Comment:
The error message should be prefixed with the function that was called, and should also wrap the error:

```suggestion
		return nil, fmt.Errorf("ed25519DecodeFromPEM: could not decode private key: %w", err)
```